packages/core/src/components/cart/CartSidebar/CartSidebar.tsx

```diff
@@ -52,38 +83,21 @@ function CartSidebar({
     icon: { icon: checkoutButtonIcon, alt: checkoutButtonIconAlt },
   },
 }: CartSidebarProps) {
-  const { currency } = useSession()
   const btnProps = useCheckoutButton()
-  const cart = useCart()
+  const { items, gifts, totalItems, isValidating, subTotal, total } = useCart()
   const { cart: displayCart, closeCart } = useUI()
   const { fadeOut } = useFadeEffect()
+  const { sendViewCartEvent } = useViewCartEvent()
 
-  const { items, gifts, totalItems, isValidating, subTotal, total } = cart
-
-  const isEmpty = items.length === 0
+  const isEmpty = useMemo(() => items.length === 0, [items])
```

Comment:
do we need this `useMemo` here?

Reply:
Yeah, it's good to memoize this value to prevent re-renders when validating cart changes that isn't related to empty the cart